### ISSUE TITLE
Add `Interactive#ask_lines` to read multiple lines?

### ISSUE THREAD

**Issue by fnordfish** [postmodern/command_kit.rb#76]
In my project, I need to interactively read multiple, long values.

I do something like this:

```ruby
stdin.gets($/ * 2, chomp: true)&.lines(chomp: true)
```

Which terminates reading on double new-line and returns either the list of lines (can be empty), or `nil` (on EOF and no input)

**Comment by postmodern** [comment]
Should it return a `String` or an `Array` of `String`s? Should the returned `String` be chomped? Should the `Array` of `Strings` each be chomped? Or should the raw input `String` be returned?

```
Prompt:
foo bar
baz qux
```

A. String

```ruby
"foo bar\nbaz qux\n"
```

B. String but chomped

```ruby
"foo bar\nbaz qux"
```

C. Array of Strings

```ruby
["foo bar\n", "baz qux\n"]
```

D. Array of chomped Strings

```ruby
["foo bar", "baz qux"]
```

**Comment by postmodern** [comment]
Also should the prompt always be followed by a newline so the input is always on it's own line, or should that be a keyword argument for that?

```
Prompt: <input here>
...
```

```
Prompt:
<input here>
...
```

**Comment by fnordfish** [comment]
D and new line after prompt. :)

**Comment by postmodern** [comment]
and should we add some text to the prompt like `(Ctrl^D to end input)` or `(enter a black line to exit)`. We could use a `case` statement to detect `nil` or `$/ * 2` terminators and craft a special message, otherwise use `"enter #{terminator.inspect} to terminate input"`.

**Comment by fnordfish** [comment]
Having something like a `terminate_with:` kwarg and generate some pre-defined help texts sounds helpful. 
`help:` could also be a arg, where on `false` it does not generate a text and gives some method of creating a custom message.

**Comment by postmodern** [comment]
@fnordfish I pushed a branch with the prototype code. It seems to be working:

```ruby
#!/usr/bin/env ruby

$LOAD_PATH.unshift(File.expand_path('../../lib',__FILE__))
require 'command_kit/command'
require 'command_kit/interactive'

class TestMultiline < CommandKit::Command

  include CommandKit::Interactive

  description 'Tests ask_multiline'

  def run
    p ask_multiline('Comment')
    p ask_multiline('Comment', terminator: $/ * 2)
  end

end

if __FILE__ == $0
  TestMultiline.start
end
```

```
$ ./examples/test_multiline.rb 
Comment (Press Ctrl^D to exit): 
foo
bar
"foo\nbar\n"
Comment (Enter two empty lines to exit): 
foo
bar

"foo\nbar\n\n"
```

Not sure how I feel about the double-newline terminator being preserved. We could `.chomp!` the terminator, but that might be inconsistent vs. reading multi-line input using `Ctrl^D` as the terminator, which returns a String ending with a single-newline.

**Comment by fnordfish** [comment]
I like it. Some thoughts:

Maybe the `help` text could be an optional kw-arg. This way you can evade issues with weird non-printable chars authors come up with. In any case, `"Enter #{terminator.inspect} to exit"` is probably saver than printing new-lines etc.

Regarding the double new lines: Maybe it's worth forwarding an optional `chomp` kw-arg to `gets`, or make it an default on double-newline?
This way, I could simply `ask_multiline('Items', terminator: $/ * 2, chomp: true).lines(chomp: true)` to get my list of items.

**Comment by postmodern** [comment]
I have updated the the default `help` messages and added specs. I feel like the help messages are worded a bit oddly.
https://github.com/postmodern/command_kit.rb/commit/3b43f4836790514938cb569b89e8005bca34f4e9

**Comment by postmodern** [comment]
@fnordfish I'm thinking of changing `ask_multiline(..., terminator: ...)` to only accept a set of finite terminator symbols (`:double_newline` and `:ctrl_d`) that then set the terminator String and help message. Should I still support accepting arbitrary terminator Strings?

**Comment by fnordfish** [comment]
No. If go that route (totally valid, IMHO), don't make it too complicated. 
_Maybe_ it's worth exploring if there could be a single method all "ask*" methods call, which takes a question string, `gets` args etc. This way your top level API stays clean and simple for the 99%, but there's still a way to get very custom behavior which still does all the stdin/stdout rebinding. IDK

**Comment by postmodern** [comment]
Circling back around to 0.6.0 work, I'm pretty confident with `ask_multiline` and will merge it.

**Comment by postmodern** [comment]
Implemented by 54e587f.
